buildSrc/build.gradle.kts

```diff
@@ -5,9 +5,9 @@ plugins {
   id("com.diffplug.spotless") version "6.25.0"
 }
 
-if (!JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_17)) {
+if (JavaVersion.current() != JavaVersion.VERSION_17) {
```

Comment:
Requiring that the current version is java 17 is a bit stricter than we need. We really need to make sure that gradle can [autodetect](https://docs.gradle.org/current/userguide/toolchains.html#sec:auto_detection) java version 17 on the system.

What about the following instead:
```
if (!hasLauncherForJavaVersion(17)) {
  throw GradleException(
    "JDK 17 is required to build and gradle was unable to detect it on the system.  " +
      "Please install it and see https://docs.gradle.org/current/userguide/toolchains.html#sec:auto_detection " +
      "for details on how gradle detects java toolchains."
  )
}

fun hasLauncherForJavaVersion(version: Int): Boolean {
  return try {
    javaToolchains.launcherFor { languageVersion = JavaLanguageVersion.of(version) }.get()
    true
  } catch (e: Exception) {
    false
  }
}
```
I tested this locally and `hasLauncherForJavaVersion` successfully returned true for versions I had installed, and false for versions I did not.